### DISCUSSION TITLE
After rabbitmq starts, it occupies 1600m memory

### DISCUSSION THREAD

**Discussion by clcc2019** [rabbitmq/rabbitmq-server#5485] [Other]
After rabbitmq starts, it occupies 1600m memory？
kubernetes v1.20.15，rabbitmq image 3.8.34

**Comment by michaelklishin** [comment]
There are diagnostics tools that [provide a breakdown of how the memory is used](https://www.rabbitmq.com/memory-use.html). [Memory allocator settings](https://www.rabbitmq.com/runtime.html#allocators) can be changed, although without knowing what exactly uses most memory, this would be pointless.

FYI, RabbitMQ 3.8 is [entirely out of support](https://www.rabbitmq.com/versions.html) and [3.8 images are no longer produced](https://github.com/docker-library/rabbitmq/pull/559) by the community. Consider moving to 3.10 on Erlang 25.

**Reply by clcc2019** [reply]
> There are diagnostics tools that [provide a breakdown of how the memory is used](https://www.rabbitmq.com/memory-use.html). [Memory allocator settings](https://www.rabbitmq.com/runtime.html#allocators) can be changed, although without knowing what exactly uses most memory, this would be pointless.
> 
> FYI, RabbitMQ 3.8 is [entirely out of support](https://www.rabbitmq.com/versions.html) and [3.8 images are no longer produced](https://github.com/docker-library/rabbitmq/pull/559) by the community. Consider moving to 3.10 on Erlang 25.

<img width="1467" alt="image" src="https://user-images.githubusercontent.com/46099295/184271606-c8b3b1fb-8352-4819-afc7-2977638d6bd0.png">
The default configuration I used and used 1.6G without the service running

**Reply by michaelklishin** [reply]
What I means was the `rabbitmq-diagnostics` command.

**Reply by michaelklishin** [reply]
3.8 is out of support => closing. 

The system part is runtime allocators, so something in this environment causes it to allocate a larger than usual block ahead of time. We have seen this on CentOS Stream 9 hosts without any specifics to make a conclusion.
